store/store.go

```diff
@@ -499,6 +506,34 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+// advanceContiguousHead updates contiguousHead and heightSub if a higher
+// contiguous header exists on a disk.
+func (s *Store[H]) advanceContiguousHead(ctx context.Context, height uint64) {
+	newHead := s.nextContiguousHead(ctx, height)
+	if newHead.IsZero() || newHead.Height() <= height {
+		return
+	}
+
+	s.contiguousHead.Store(&newHead)
+	s.heightSub.SetHeight(newHead.Height())
+	log.Infow("new head", "height", newHead.Height(), "hash", newHead.Hash())
+	s.metrics.newHead(newHead.Height())
+}
+
+// nextContiguousHead returns a next contiguous header if any.
+func (s *Store[H]) nextContiguousHead(ctx context.Context, height uint64) H {
+	var newHead H
+	for {
+		height++
```

Comment:
Should we fetch just `height` first to see that the header exists at all and start looking for the next after?

No idea, just have realised that there is a window of opportunity that we might look for a contiguous headers when initial doesn't exist (which sounds like a disk datastore corruption or dunno).